src/main/java/spoon/template/TemplateMatcher.java

```diff
@@ -62,12 +62,9 @@
 public class TemplateMatcher implements Filter<CtElement> {
 
 	/**
-	 * Collects all AST nodes, which has to be substituted, because they represents a template parameter declared by {@link TemplateParameter}
+	 * Searches for all invocations of {@link TemplateParameter#S()} in "root", a CtClass model of {@link Template}
 	 *
-	 * Searches for all invocations of {@link TemplateParameter#S()} in CtClass model of {@link Template}
 	 * @param root CtClass model of {@link Template}
-	 * @return List of all variables - only these parameters which are declared by {@link TemplateParameter}.
-	 * It doesn't includes references to parameters annotated by {@link Parameter}
 	 */
```

Comment:
Your new comment is absolutely correct, I just suggest to keep my comment too (in improved form), because it explains purpose of this method. 
It is used like this:
```java
public TemplateMatcher(CtElement templateRoot) {
  ...
  variables = getMethods(templateType)
  ...
}
```
so it returns template variables/parameters/substitution points (please give it correct name ;-) ) ... but as written on last javadoc line, they are not all, because variables/parameters/substitution points  annotated by {@link Parameter} are not included in returned list